engine/runtime/src/main/scala/org/enso/compiler/pass/analyse/AmbiguousImportsAnalysis.scala

```diff
@@ -0,0 +1,430 @@
+package org.enso.compiler.pass.analyse
+
+import org.enso.interpreter.runtime.Module
+import org.enso.compiler.context.{InlineContext, ModuleContext}
+import org.enso.compiler.core.IR
+import org.enso.compiler.core.IR.Module.Scope.Import
+import org.enso.compiler.core.IR.Module.Scope.Import.Polyglot
+import org.enso.compiler.data.BindingsMap
+import org.enso.compiler.exception.CompilerError
+import org.enso.compiler.pass.IRPass
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/** A pass that checks for ambiguous and duplicated symbols from imports.
+  * A duplicated import is an import of a symbol that has already been imported and refers to the
+  * same object (entity). On the other hand, an ambiguous import is an import of a symbol that has already
+  * been imported but refers to a different object. For every duplicated import, a warning is attached
+  * to the IR, and for every ambiguous import, the IR is replaced with an error.
+  * To identify an object, this pass uses physical path of the object instead of the object itself.
+  *
+  * One import IR can be replaced with multiple error IRs. This is the case for `from ... import ...`
+  * import statements.
+  *
+  * The original import is saved in the error and warning so that the user can see from which location
+  * the symbol was originally imported.
+  *
+  * Also iterates polyglot imports.
+  *
+  * All synthetic imports and exports, as well as synthetic modules are ignored by this pass.
+  *
+  * Thi pass does not alter any metadata.
+  */
+case object AmbiguousImportsAnalysis extends IRPass {
```

Comment:
It'd be better, if we wrote the new code in Java. Inspiration is [available here](https://github.com/enso-org/enso/pull/6905/files#diff-b032a7bebd3f2aab7b0542150206b9ae4997efd8dff9a826986ffdae9d97615bR32). Btw. yes, I do realize interacting with Scala from Java isn't easy, but it is not easy to maintain (more of) Scala code either.